src/libs/pool-data-provider/hooks/use-incentives-data-context.tsx

```diff
@@ -0,0 +1,292 @@
+            reserveUnderlyingAddress = '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270';
+          } else if (reserve.symbol === 'ETH') {
+            reserveUnderlyingAddress = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
+          } else if (reserve.symbol === 'AVAX') {
+            reserveUnderlyingAddress = '0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7';
+          }
+        }
+        // Find the corresponding incentive data for each reserve
+        const incentiveData = reserveIncentiveData.find(
+          (incentive) => incentive.underlyingAsset.toLowerCase() === reserveUnderlyingAddress
+        );
+        if (incentiveData) {
+          const calculatedReserveIncentives: CalculateReserveIncentivesResponse =
+            calculateReserveIncentives({
+              reserveIncentiveData: incentiveData,
+              totalLiquidity: formatBNInput(reserve.totalLiquidity, Number(reserve.decimals)),
+              liquidityIndex: formatBNInput(reserve.liquidityIndex, RAY_DECIMALS),
+              totalScaledVariableDebt: formatBNInput(
+                reserve.totalScaledVariableDebt,
+                Number(reserve.decimals)
+              ),
+              totalPrincipalStableDebt: formatBNInput(
+                reserve.totalPrincipalStableDebt,
+                Number(reserve.decimals)
+              ),
+              priceInMarketReferenceCurrency: formatBNInput(reserve.price.priceInEth, ETH_DECIMALS), // Will be replaced with marketReferencePrice/Decimals
+              decimals: Number(reserve.decimals),
+
+              aRewardTokenPriceInMarketReferenceCurrency: calculateRewardTokenPrice(
+                reserves,
+                incentiveData.aIncentiveData.rewardTokenAddress.toLowerCase(),
+                incentiveData.aIncentiveData.priceFeed,
+                incentiveData.aIncentiveData.priceFeedDecimals
+              ),
+
+              vRewardTokenPriceInMarketReferenceCurrency: calculateRewardTokenPrice(
+                reserves,
+                incentiveData.vIncentiveData.rewardTokenAddress.toLowerCase(),
+                incentiveData.vIncentiveData.priceFeed,
+                incentiveData.vIncentiveData.priceFeedDecimals
+              ),
+              sRewardTokenPriceInMarketReferenceCurrency: calculateRewardTokenPrice(
+                reserves,
+                incentiveData.sIncentiveData.rewardTokenAddress.toLowerCase(),
+                incentiveData.sIncentiveData.priceFeed,
+                incentiveData.sIncentiveData.priceFeedDecimals
+              ),
+            });
+          calculatedReserveIncentives.underlyingAsset = isBaseAsset
+            ? '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+            : calculatedReserveIncentives.underlyingAsset; // ETH, MATIC, and AVAX all use reserve address of '0xee...'
+          allReserveIncentives.push(calculatedReserveIncentives);
+        }
+      });
+      setReserveIncentives(allReserveIncentives);
+    }
+  }, [reserves, reserveIncentiveData]);
```

Comment:
stringify it our combine with the above effect and just calculate based on an interval instead if reacting to input.